cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/mapping/CfArgumentsCreator.java

```diff
@@ -101,4 +125,47 @@
         return args.toArray(new String[0]);
     }
 
+    /**
+     * Extends the commandLine with the missing options by reading the given yaml
+     * file.
+     *
+     * @param missingOptions missing Options like -s, -a, -o
+     * @param args           Commandline arguments
+     * @return Commandline arguments
+     */
+    private static String[] extendForDiffAndApplyCommand(List<String> missingOptions, LinkedList<String> args) {
+        try {
+            String yamlPath = args.get(args.indexOf("-y") + 1);
+            ConfigBean configBean = YamlMapper.loadBeanFromFile(yamlPath, ConfigBean.class);
+            TargetBean targetBean = configBean.getTarget();
+
+            missingOptions.forEach(key -> {
+                args.add(key);
+                String value = "";
+                switch (key) {
+                case "-a":
+                    value = targetBean.getEndpoint();
+                    break;
+                case "-o":
+                    value = targetBean.getOrg();
+                    break;
+                case "-s":
+                    value = targetBean.getSpace();
+                    break;
+                default:
+                    break;
+                }
+
+                args.add(value);
+                log.info("Extended CommandLine Argument with the Option: " + key +
+                    " and value " + "'" + value + "'");
+            });
+        } catch (IOException | ParserException | ScannerException ex) {
```

Comment:
The error handling for `RefResolvingException` and `ConstructorException` is still missing. Also the entire try-block can be made smaller such that it only contains the `YamlMapper.loadBeanFromFile(...)` call, beacuse this is the only place where these exceptions might be thrown